### DISCUSSION TITLE
Goggles 2 iNav 6.1 and full softserial

### DISCUSSION THREAD

**Discussion by ifonlyitwaseasy** [iNavFlight/inav-configurator#1810] [Q&A] [answered]
Goggles2 OSD with Soft serial
I have used DJI with WTFOS and stock on a number of 411SE and 411wing FCs using full softserial port without any issues. I now have DJI Goggles2 and can't get the OSD to work on the FC using an original air unit when I have upgraded the firmware to the latest firmware and bound to the goggles2. I have set port to MSPdisplayport and am using BF43COMPAT.

**Comment by DzikuVx** [comment, marked as answer]
Running OSD on softwareserial is not recommended and not guaraneed to work

**Reply by ifonlyitwaseasy** [reply]
Paweł many thanks for the reply.  I was quite happy with the original air units and vistas on soft serial but take your on board your advise regarding soft serial.  The 411SE and 411 Wing just have 2 uarts and a soft serial (duplex if flashed).  This means that they are no use for DJI use unless I can use the SS for the gps?  I know it isn't recommended but as I use it for guidance on direction to home and am not after rapid updates is it OK to use SS for GPS or are these units now redundant?

**Reply by MrD-RC** [reply]
It’s been recommended not to get F411 or F722 flight controllers for over 2 years now. This is one of the reasons why. The other is they are running out of space. Of course, already having these FCs is fair enough. But they have limitations.

**Reply by ifonlyitwaseasy** [reply]
Darren, I know it isn't ideal but what are the limitations of GPS on SS?

**Comment by ifonlyitwaseasy** [comment]
Darren

Thanks for the reply. I have had these in planes for well over 2 years
working with dji air and vistas  first stock and then with WTFOS.  I won't
buy any more BUT they still work. Also pulling them out is tricky when
already mounted on 3d frames that fit. I can replace but was just joking
not to.

Regards

Neil
